File: cpp/6.cpp
#include<iostream>
#include<vector>
using namespace std;

class Solution 
{
public:
    string convert(string s, int numRows) 
    {
        if(numRows==1)
            return s;
        vector<vector<char>> st;
        int length=s.length();
        int pt=0;
        for (int i = 0;; i++)
        {
            vector<char>t;
            for (int j = 0; j < numRows; j++)
            {
                if(i%(numRows-1)==0||(i%(numRows-1)==numRows-j-1))
                    t.push_back(s[pt++]);
                else 
                    t.push_back(' ');
                if(pt == length)
                    break;
            }
            st.push_back(t);
            if(pt == length)
                    break;
        }
        string re;
        for (int i = 0; i < numRows; i++)
        {
            for (int j = 0; j < st.size(); j++)
            {
                if(st[j][i]!=' '&&st[j][i]!='\0')
                    re.push_back(st[j][i]);
            }
        }
        return re;
    }
};

File: py/6.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:

        array = [s for i in range(numRows)]

        resArray = [list() for i in range(numRows)]

        i, d = 0, 1

        for k in range(len(s)):
            resArray[i].append(array[i][k])

            i += d

            if i == 0 or i == numRows - 1:
                d = d * (-1)

        res = ""
        print(resArray)

        result = list()
        for r in resArray:
            result.extend(r)
        for l in resArray:
            res += "".join(l)

        return "".join(result)

if __name__ == '__main__':
    solu = Solution()

    print(solu.convert(s = "LEETCODEISHIRING", numRows = 4))